src/services/formatting/formatting.ts

```diff
@@ -387,7 +389,10 @@ namespace ts.formatting {
 
             let indentation = inheritedIndentation;
             if (indentation === Constants.Unknown) {
-                if (isSomeBlock(node.kind)) {
+                if (isIndentPreventedChildNode(parent.kind, node.kind)) {

```

Comment:
this is similar to closing braces and parens, maybe we should add it to get indentation in token.